datamap/receiver.go

```diff
@@ -0,0 +1,473 @@
+type bareBase struct {
+	Type                ItemType `json:"type"`
+	Hash                string   `json:"hash"`
+	ContentID           int      `json:"content_id"`
+	ContentName         string   `json:"content_name"`
+	Title               string   `json:"title"`
+	CreationDate        string   `json:"creation_date"`
+	PublicationDate     string   `json:"publication_date"`
+	NavContext          []string `json:"navigation_context"`
+	ValidFrom           string   `json:"valid_from"`
+	ValidTo             string   `json:"valid_to"`
+	AnalyticsCategory   string   `json:"analytics_category"`
+	AdvertisingCategory string   `json:"advertising_category"`
+	StationName         string   `json:"station_name"` //new api adition that seems to be set/get locally
+}
+
+func (b *bareBase) unmarshalBareBase(r *Receiver) {
+	b.Hash = r.Hash
+	b.ContentID = r.ContentID
+	b.ContentName = r.ContentName
+	b.Title = r.Title
+	b.PublicationDate = r.PublicationDate
+	b.CreationDate = r.CreationDate
+	b.NavContext = r.NavContext
+	b.ValidFrom = r.ValidFrom
+	b.ValidTo = r.ValidTo
+	b.AnalyticsCategory = r.AnalyticsCategory
+	b.AdvertisingCategory = r.AdvertisingCategory
+}
+
+// this struct here is a list of the most common properties shared
+// among some content types (the barebone properties are in bareBase)
+type fullBase struct {
+	bareBase
+
+	TeaserTitle       string        `json:"teaser_title"`
+	TeaserText        string        `json:"teaser_text"`
+	TeaserImage       string        `json:"teaser_image"`
+	Author            string        `json:"author"`
+	Authors           []Person      `json:"author_objects"`
+	EditorialComment  string        `json:"editorial_comment"`
+	Copyright         string        `json:"copyright"`
+	Copyrights        []Copyright   `json:"copyright_objects"`
+	Media             []Item        `json:"media"`
+	CanonicalURL      string        `json:"canonical_url"`
+	URL               string        `json:"url"`
+	Categories        []interface{} `json:"categories"`
+	Struct            []interface{} `json:"struct"`
+	CommentingEnabled bool          `json:"commenting_enabled"`
+	NoFollow          bool          `json:"no_follow"`
+	NotSearchable     bool          `json:"not_searchable"`
+	Period            string        `json:"period"`
+	Keywords          string        `json:"keywords"`
+}
+
+func (f *fullBase) unmarshalFullBase(r *Receiver) error {
+	f.unmarshalBareBase(r)
+
+	f.TeaserTitle = getTeaserTitle(r)
+	f.TeaserText = r.TeaserText
+	f.TeaserImage = r.TeaserImage
+	f.Author = r.Author
+	f.Authors = r.Authors
+	f.EditorialComment = r.EditorialComment
+	f.Copyright = r.Copyright
+	f.CanonicalURL = r.CanonicalURL
+	f.URL = r.URL
+	f.Categories = r.Categories
+	f.Struct = r.Struct
+	f.CommentingEnabled = r.CommentingEnabled
+	f.NoFollow = r.NoFollow
+	f.NotSearchable = r.NotSearchable
+	f.Period = r.Period
+	f.Keywords = r.Keywords
+	medias, err := unmarshalItems(r.Media)
+	f.Media = medias
+
+	return err
+}
+
+// Map represents a map
```

Comment:
so, starting at this line are types that only appear embedded in other types? i think it's a bit confusing to have them in the same file as the receiver stuff, maybe put them in a separate file? totally minor though and i might be misunderstanding what they do or why they're here